src/__tests__/pages/api/auth/reset-password.test.ts

```diff
@@ -0,0 +1,299 @@
+import dotenv from 'dotenv';
+dotenv.config();
+
+import {
+  mockUserData,
+  mockPassword,
+  mockEmail
+} from '../../../../__mocks__/pages/api/auth';
+import signUpHandler from '../../../../pages/api/auth/signup';
+import resetPasswordHandler from '../../../../pages/api/auth/reset-password';
+import { PrismaClient } from '@prisma/client';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { hashFunction } from '../../../../utils/password';
+import { resetPasswordProps } from '../../../../types/api/auth/reset-password';
+
+const prisma = new PrismaClient();
+
+let email: string, password: string;
+
+beforeEach(async () => {
+  const userData = mockUserData();
+  email = userData.email;
+  password = userData.password;
+
+  const req = {
+    body: {
+      email,
+      password
+    }
+  } as NextApiRequest;
+
+  const json = jest.fn();
+
+  const status = jest.fn(() => {
+    return { json };
+  });
+
+  const res = {
+    status
+  } as unknown as NextApiResponse;
+
+  await signUpHandler(req, res);
+});
+
+describe('Successfully change password', () => {
+  it('Change password successs', async () => {
+    const hashPassword = (
+      await prisma.user.findUnique({
+        where: {
+          email
+        },
+        select: {
+          password: true
+        }
+      })
+    ).password;
+
+    const newPassword = mockPassword();
+
+    const req = {
+      body: {
+        email: email,
+        oldPassword: password,
+        newPassword: newPassword
+      } as resetPasswordProps
+    } as NextApiRequest;
+
+    const json = jest.fn();
+
+    const status = jest.fn(() => {
+      return { json };
+    });
+
+    const res = {
+      status
+    } as unknown as NextApiResponse;
+
+    await resetPasswordHandler(req, res);
+
+    expect(status.mock.calls[0]).toEqual([200]);
+    expect(json.mock.calls[0]).toEqual([
+      {
+        message: 'Password changed'
+      }
+    ]);
+
+    const currentPassword = (
+      await prisma.user.findUnique({
+        where: {
+          email
+        },
+        select: {
+          password: true
+        }
+      })
+    ).password;
+
+    expect(hashPassword).not.toEqual(currentPassword);
+    expect(hashFunction(newPassword)).toEqual(currentPassword);
+  });
+});
+
+describe('Fail to change password', () => {
+  it('Wrong oldPassword', async () => {
+    const req = {
+      body: {
+        email: email,
+        oldPassword: 'wrongPassword',
```

Comment:
Hmmm but technically it's possible for this to actually be the right password though, right? Maybe do like `'wrongPassword' + password` to ensure it's always wrong

Reply:
since the hash function is working (by unit testing the hash function), we can assume all password has the same length = 32 or 64 or sth i forgot haha

Reply:
But when we send `oldPassword` to the API endpoint, it'll hash it anyway and compare by hashed value. Technically, the original unhashed password could be `'wrongPassword'` since it's random, right? If yes, then we need a way to make sure

Reply:
oh yes!